### PR TITLE
Add HttpClient typespec opt

### PR DESCRIPTION
**Pull request by rupurt**

### CODE REVIEW

File: lib/ethereumex/http_client.ex [hunk 1]
```diff
@@ -4,7 +4,9 @@ defmodule Ethereumex.HttpClient do
   use Ethereumex.Client.BaseClient
   alias Ethereumex.Config
 
-  @spec post_request(binary(), []) :: {:ok | :error, any()}
+  @type opt :: {:url, String.t()}
```
Comment (ayrat555) [inline lib/ethereumex/http_client.ex:7]: it also can be empty. 

I think the correct typespec for opts would be
```elixir
[] | [url: String.t()] 
```

**Review by ayrat555 — CHANGES_REQUESTED** [review]

File: lib/ethereumex/http_client.ex (hunk 1 above)
Comment (rupurt) [reply lib/ethereumex/http_client.ex:7]: I'm happy to change it to your suggestion. But from my understanding, both my current approach and your suggested example are valid. My current approach also covers the case of future unordered options.

https://hexdocs.pm/elixir/typespecs.html#literals

Comment (ayrat555) [reply lib/ethereumex/http_client.ex:7]: sounds good. just add braces. and pr's good to go

Comment (rupurt) [reply lib/ethereumex/http_client.ex:7]: :+1: updated with braces

**Pull request merged**